pkg/controller/podgc/gc_controller.go

```diff
@@ -50,6 +52,15 @@ const (
 	quarantineTime = 40 * time.Second
 )
 
+const (
+	// Pod condition indicating the pod is about to be deleted due to an external
+	// disruption (such as preemption, eviction API or garbage-collection).
+	// The constant is not exported in order to prevent its external use until
+	// the name is accepted within the KEP-3329. Once the name is accepted the
+	// declaration will be moved to the core API.
+	targetedForDisruption v1.PodConditionType = "TargetedForDisruption"
```

Comment:
I saw the old comment about not putting this in a central file until it's beta but I'm not convinced declaring it in N places is better.

The primary problem about exposing it is that then people might use it and it would be hard to change the name. But I don't think multiple declarations solves that problem at all.

Reply:
My thinking is that it solves the problem of using the pod condition directly from code at compile time without any notification. Still, one could use the pod condition at runtime, but needs to enable the feature gate flag in order to trigger appending of these pod conditions. Multiple declaration places could be hard to maintain in the long run, but the idea is that we eliminate the duplication in Beta. Any other ideas?

Reply:
As for the name, I like `DisruptionTarget`, @alculquicondor ?

Reply:
> solves the problem of using the pod condition directly from code at compile time without any notification

You could make the name of the constant AlphaWillBeRenamedNoCompatGuaranteeDisruptionTargetCondition or something like that.

Reply:
Indeed, it avoids the duplication and  will prevent users for committing to the name, but it may look a little bit awkward in the core API.

I'm personally fine with both approaches:
1. DisruptionTarget duplicated is each package for Alpha
2. using the proposed name, but I'm thinking about `DisruptionTargetToBeRenamedForBeta` it seems clear enough that if it is to be renamed, then no compat guarantee

Any further opinions?

Reply:
Personally I prefer option 2, @liggitt may have a different opinion though

Reply:
Re name: The guideline is the following (from https://github.com/kubernetes/community/blob/308bb1109fe07cf3ecf14381129ecabac98de425/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties):

> Condition type names should describe the current observed state of the resource, rather than describing the current state transitions. This typically means that the name should be an adjective ("Ready", "OutOfDisk") or a past-tense verb ("Succeeded", "Failed") rather than a present-tense verb ("Deploying"). 

I think both qualify, then favoring the shorter `DisruptionTarget` should be fine.

I actually like `AlphaWillBeRenamedNoCompatGuaranteeDisruptionTargetCondition`... very in-your-face. But maybe something like `AlphaNoCompatGuaranteeDisruptionTargetCondition` and a comment should suffice. @liggitt are you ok with this or would you rather have the variables private?

Reply:
For now I have updated the private pod condition name from `TargetedForDisruption` to `DisruptionTarget`. I will wait for the @liggitt opinion and general consensus before introducing the in-face name into the core API. I'm ok with `AlphaNoCompatGuaranteeDisruptionTargetCondition` as the name, but I think the last `Condition` word is redundant, cause this is conveyed by the type already.

Reply:
Knowing how our API packages get consumed, I really didn't want to export things in from core/v1 that aren't stable (either stable go field names that represent unstable functionality, or unstable names we intend to remove/break callers of in the future). That's not an official API guidelines position, just my $0.02 having heard friction reports of people using our libraries.

If other API folks think explicitly unstable constant names are reasonable, I won't block on that. I'd probably recommend a simple `Alpha...` prefix to the go field name and explanatory godoc sufficient in that case.

Reply:
We intend to break go compatibility rules, and not the k8s API rules. That's why I suggested an especially unwieldy name. I went a bit overboard, but I'd use more than just "Alpha" to warn people.

I don't have a super strong opinion, I just wouldn't say that copy-paste inheritance is clearly better than breaking compatibility rules.

Reply:
I have pushed a commit which introduces the new pod condition type into the core APIs (I make use of the new constants in the next commit):
https://github.com/kubernetes/kubernetes/pull/110959/commits/c96585e316f3f83db98462ff4876ff5235fede3f

A couple of notes:
1. I've reviewed how other pod condition types were introduced. For reference:
- ContainersReady (2018)
  - Commit: https://github.com/kubernetes/kubernetes/commit/6b08ef575fe52f49acac53583d6b954ad27a8806
  - PR: https://github.com/kubernetes/kubernetes/pull/64646
  - Approach: Direct introduction into into core API
- Unschedulable (2016)
  - Commit: https://github.com/kubernetes/kubernetes/commit/859077b70183993f492fd2d3de66b9effcd6fef1
  - PR:https://github.com/kubernetes/kubernetes/pull/34634
  - Approach: Replaced a string
- Initialized (2016)
  - Commit: https://github.com/kubernetes/kubernetes/commit/6685715c4ca1f143108273edd038806b2bb5fd72
  - PR: https://github.com/kubernetes/kubernetes/pull/23567
  - Approach: Direct introduction into core API
- PodScheduled (2016)
  - Commit: https://github.com/kubernetes/kubernetes/commit/a80b1798c45f88e6b38261a1657c0dfa787de136
  - PR: https://github.com/kubernetes/kubernetes/pull/24459
  - Approach: Direct introduction into core API

2. I shortened the name and removed the `Condition` word from the end as the other pod condition types don't have it

3. I keep the string value represented by the constant as `DisruptionTarget`, but can update. Any opinions?